rfcs/20240311-verbose-logging-support/README.md

```diff
@@ -0,0 +1,52 @@
+# Proposal for a Logging Mechanism in oneDNN
+
+## Motivation
+
+This is a proposal to introduce a logging mechanism within the oneDNN framework to direct, save and manage the verbose outputs generated from oneDNN into user-specified logs.
+Presently, oneDNN in the verbose mode prints all information directly to the console using `stdout` - with a logging mechanism, the user will be able to create custom logs by controlling what and how verbose information is saved in the user-specified logfiles. 
+
+## Proposal
+The proposal is to implement logging support in oneDNN with the help of [`spdlog`]((https://github.com/gabime/spdlog)), a header-only C++ logging library which provides a Python-like formatting API using the bundled [fmt](https://github.com/fmtlib/fmt) lib. The library uses an MIT license, has no cross-component dependencies and has a distinct performance [gain](https://github.com/gabime/spdlog/tree/v1.x?tab=readme-ov-file#benchmarks) over `sprintf` for logging. Key considerations for the proposal are listed as follows:
+
+### 1. Build Options
+`spdlog` can be built with oneDNN using a header-only approach by adding the library headers to the build tree. A pre-compiled version is recommended by the authors to reduce compilation overhead but is not required since the implementation will utilize only limited features from the entire scope of `spdlog`. 
+
+### 2. Runtime Logging Controls
+A basic requirement for implementing logging support will be to define the environmental control variables which the user can specify to manage oneDNN data logging. 
+For the simple case where the logging mechanism involves directly dumping the verbose outputs into a logfile, this can be accomplished with two control variables, one for enabling the data logging (`ENABLE_ONEDNN_LOGGING`) and the other for specifying the logfile path (`ONEDNN_LOGFILE=/path/to/file`).
```

Comment:
Could you clarify why is `ENABLE_ONEDNN_LOGGING` needed?

IIUC, we are switching to spdlog internally for all prints. Based on that, I would expect that only `ONEDNN_VERBOSE_FILE` is needed, with stdout its default value.

Also, environment variable typically need a matching API.

Reply:
Good point! The specification of a logfile path should be enough to enable logging in oneDNN. Updating RFC accordingly.